cl/users/tests.py

```diff
@@ -2740,3 +2741,279 @@ def test_retry_datetime(self, mock_send_email) -> None:
+        message = json.loads(notification["Message"])
+        mail_obj = message.get("mail")
+        event_obj = message.get(event_name, {})
+
+        # Send signal
+        signal_kwargs = dict(
+            sender=self,
+            mail_obj=mail_obj,
+            raw_message=raw,
+        )
+        signal_kwargs[f"{event_name}_obj"] = event_obj
+        signal.send(**signal_kwargs)
+
+    def test_multiple_bounce_subtypes(self) -> None:
+        """This test checks if we can assign properly the bounce subtype for
+        EmailFlag and BackoffEvent objects
+        """
+
+        # Trigger a hard bounce notification event
+        self.send_signal(
+            self.no_email_hard_bounce_asset, "bounce", signals.bounce_received
+        )
+        email_ban = EmailFlag.objects.filter(
+            email_address="bounce@simulator.amazonses.com",
+            object_type=OBJECT_TYPES.BAN,
+        )
+        # The bounce subtype should be NOEMAIL
+        self.assertEqual(email_ban[0].event_sub_type, SUB_TYPES.NOEMAIL)
+
+        # Trigger a soft bounce notification event
+        self.send_signal(
+            self.mail_box_full_soft_bounce_asset,
+            "bounce",
+            signals.bounce_received,
+        )
+        backoff_event = BackoffEvent.objects.filter(
+            email_address="bounce@simulator.amazonses.com",
+        )
+        # The bounce subtype should be MAILBOXFULL
+        self.assertEqual(
+            backoff_event[0].event_sub_type, SUB_TYPES.MAILBOXFULL
+        )
+
+    def test_broken_email_address_banner(self) -> None:
+        """This test checks if soft and hard bounces events properly trigger a
+        broken email address banner, a Permanent broken email address banner
+        overrides a previous Transient broken banner.
+        """
+
+        path = reverse("show_results")
+        # An anonymous user should never see the broken email banner.
+        r = self.client.get(path)
+        self.assertEqual(r.context["EMAIL_BROKEN_BANNER"][0], False)
+
+        user = self.user
+        user.email = "bounce@simulator.amazonses.com"
+        user.password = make_password("password")
+        user.save()
+        # Authenticate user
+        login = self.client.login(username=user.username, password="password")
+        r = self.client.get(path)
+
+        # The user's email has no problems, no banner showed
+        self.assertEqual(r.context["EMAIL_BROKEN_BANNER"][0], False)
+
+        # Trigger a small_email_only flag
+        self.send_signal(
+            self.msg_large_bounce_asset, "bounce", signals.bounce_received
+        )
+        email_ban = EmailFlag.objects.filter(
+            email_address="bounce@simulator.amazonses.com",
+            object_type=OBJECT_TYPES.BAN,
+        )
+        self.assertEqual(email_ban.count(), 0)
+        r = self.client.get(path)
+        # An small_email_only EmailFlag object shouldn't trigger a broken
+        # email banner
```

Comment:
Im spelunking through the code and admin panel and I have no idea what 

small_email_only means.

Reply:
sure, `small_email_only` is a kind of `EmailFlag` object (the other type is a Ban) that we use to flag an email address from we received a "MessageTooLarge" or "AttachmentRejected" soft bounce notification, so email addresses flagged with `small_email_only` will continue receiving messages, but we'll only send emails without attachments.